test/linalg/diagonal.jl

```diff
@@ -312,7 +312,7 @@ end
 end
 
 # allow construct from range
-@test Diagonal(linspace(1,3,3)) == Diagonal([1.,2.,3.])
+@test Diagonal(linspace(1,3,3)) ≈ Diagonal([1.,2.,3.])
```

Comment:
Right. As you say, that's a thorny issue... ;-) Until then, better use `all` indeed.